service/src/main/kotlin/no/nav/su/se/bakover/service/regulering/ReguleringServiceImpl.kt

```diff
@@ -227,21 +233,43 @@ class ReguleringServiceImpl(
             KunneIkkeFerdigstilleOgIverksette.KunneIkkeBeregne
         }
             .flatMap { beregnetRegulering ->
-                beregnetRegulering.simuler(utbetalingService::simulerUtbetaling)
-                    .mapLeft {
-                        log.error("Regulering for saksnummer ${regulering.saksnummer}. Simulering feilet.")
-                        KunneIkkeFerdigstilleOgIverksette.KunneIkkeSimulere
-                    }.flatMap {
-                        if (it.simulering!!.harFeilutbetalinger()) {
-                            log.error("Regulering for saksnummer ${regulering.saksnummer}: Simuleringen inneholdt feilutbetalinger.")
-                            KunneIkkeFerdigstilleOgIverksette.KanIkkeAutomatiskRegulereSomFørerTilFeilutbetaling.left()
-                        } else {
-                            it.right()
-                        }
+                beregnetRegulering.simuler { beregning, uføregrunnlag ->
+                    sak.lagNyUtbetaling(
+                        saksbehandler = beregnetRegulering.saksbehandler,
+                        beregning = beregning,
+                        clock = clock,
+                        utbetalingsinstruksjonForEtterbetaling = UtbetalingsinstruksjonForEtterbetalinger.SammenMedNestePlanlagteUtbetaling,
+                        uføregrunnlag = uføregrunnlag,
+                    ).let {
+                        sak.simulerUtbetaling(
+                            utbetalingForSimulering = it,
+                            periode = regulering.periode,
+                            simuler = { utbetalingForSimulering: Utbetaling.UtbetalingForSimulering, periode: Periode ->
+                                utbetalingService.simulerUtbetaling(
+                                    utbetalingForSimulering,
+                                    periode,
+                                )
+                            },
+                            kontrollerMotTidligereSimulering = regulering.simulering,
+                            clock = clock,
+                        )
+                    }.map { simulertUtbetaling ->
+                        simulertUtbetaling.simulering
+                    }
```

Comment:
Generell kommentar som gjelder for alle bruk av `sak.lagNyUtbetaling()` og `sak.simulerUtbetaling()`: Det føles litt rart og unødvendig å sende med informasjon som saken burde kjenne til inn til saken igjen (les; beregning, periode, uføregrunnlag for en søk/rev/reg) for å få gjort simuleringen - ideelt sett kunne nok dette vært refaktorert videre til man sto igjen med at man kun kalte `sak.simulerUtbetaling("id til behandling som skal simuleres")` og overlot resten til saken. For å få til noe sånt kreves det at saken alltid er ajour med behandlingen som foreligger (sak må inneholde den "nyeste" versjonen av behandlingen) - p.t brukes det hovedsaklig objekter som ikke eksistere i sin nyeste form på saken (unntaket er vel simulering av søknadsbehandling) og jeg har derfor valgt å ikke ta denne refaktoreringen lenger enn hit.